linux_os/guide/system/accounts/accounts-restrictions/password_expiration/accounts_password_set_max_life_existing/oval/shared.xml

```diff
@@ -21,16 +21,21 @@
   <!-- Define a second test to ensure the maximum password life is at least the defined
        minimum (usually 1). -->
   <unix:shadow_test id="test_password_max_life_existing_minimum" version="1"
-    check="all" check_existence="at_least_one_exists"
+    check="all" check_existence="any_exist"
     comment="Password maximum life entry is at least a defined minimum">
     <unix:object object_ref="object_shadow_password_users_max_life_existing"/>
     <unix:state state_ref="min_max_password_change_interval"/>
   </unix:shadow_test>
 
   <unix:shadow_object id="object_shadow_password_users_max_life_existing" version="1">
     <unix:username operation="pattern match">.*</unix:username>
+    <filter action="include">non_system_accounts_max_life</filter>
   </unix:shadow_object>
 
+  <unix:shadow_state id="non_system_accounts_max_life" version="1">
+      <unix:password operation="pattern match">[^\!\*]</unix:password>
```

Comment:
I have tested this locally and the `correct_max_pas_age.pass.sh` fails when the user has a password defined, but was temporarily locked. For example:
`root:!!$6$dCikIu1...::19200:0:99999:7:::`

Maybe this regex could be aligned to the remediation by using this:
`^[^\!\*]+$`

Reply:
It would also be nice to have a test scenario for this case.

Reply:
this raises the question if we should care about locked accounts?

Also could you share more about the test failure, because it shouldn't fail because of that.
both remediation and audit won't consider the locked account with the changes proposed here, therefore the test must be failing because of something else.

Reply:
Good question. I see the "locking" feature mostly as something temporary, usually done by admins or based on aging criteria. In this context, I believe it is valid to check them too since there is a implicit expectation that this account can be unlocked in the future. On the other hand, this cases are already ignored by the remediations. So, I would recommend to keep the remediation as they are for now and just update the regex in this OVAL pattern to be aligned to the remediation.
It would be nice to open a [discussion](https://github.com/ComplianceAsCode/content/discussions) for this.

Sure, I executed the tests with `automatus`.
I this case, I simply locked the root account and tested the rule. I saw the root account was not remediated but was included in the list of OVAL objects with that regex defined in the pattern. My suggestion was to update the OVAL so locked users are also ignored in OVAL.

Reply:
oh now I understood what you meant with the failure, thanks!
I've pushed a fix for it and the filter id.